### PR TITLE
Add for_lane and for_platform documentation

### PR DESCRIPTION
**Pull request by mfurtak**
Documents https://github.com/fastlane/fastlane/pull/7859

### CODE REVIEW

File: docs/advanced.md [hunk 1]
```diff
@@ -337,3 +337,52 @@ Credential felix@krausefx.com:********* added to keychain.
 fastlane-credentials remove --username felix@krausefx.com
 password has been deleted.
 ```
+
+# Control configuration by lane and by platform
+
+In general, configuration files take only the first value given for a particular configuration item. That means that for an `Appfile` like the following:
+
+```ruby
+app_identifier "com.used.id"
```
Comment (KrauseFx) [inline docs/advanced.md]: Until now, I think that's how it happens, however I don't think we ever documented or added tests for the order of those. Maybe by saying that's how it works, we should have at least one automated test for this behaviour (I don't think we have one currently)

File: docs/advanced.md [hunk 2]
```diff
@@ -337,3 +337,52 @@ Credential felix@krausefx.com:********* added to keychain.
...
+the `app_identfier` will be `"com.used.id"` and the second value will be ignored. The `for_lane` and `for_platform` configuration blocks provide a limited exception to this rule.
+
+All configuration files (Appfile, Matchfile, Screengrabfile, etc.) can use `for_lane` and `for_platform` blocks to control (and override) configuration values for those circumstances.
+
+`for_lane` blocks will be called when the name of lane invoked on the command line matches the one specified by the block. So, given a `Screengrabfile` like:
+
+```ruby
+locales ['en-US', 'fr-FR', 'ja-JP']
+
+for_lane :screenshots_english_only do
```
Comment (KrauseFx) [inline docs/advanced.md]: 👍

File: docs/advanced.md (hunk 1 above)
Comment (mfurtak) [inline docs/advanced.md]: We do have tests for this. 👍 

https://github.com/fastlane/fastlane/blob/master/fastlane_core/spec/fixtures/ConfigFileRepeatedValueSet

This is what I was able to restore after reworking that original PR.

Comment (KrauseFx) [inline docs/advanced.md]: @mfurtak Beautiful, thanks for confirming 👍

**Pull request merged**